### DISCUSSION TITLE
Clarify CockroachDB status vs PostgreSQL

### DISCUSSION THREAD

**Discussion by hexmode** [zitadel/zitadel#8774] [Q&A] [answered]
### Preflight Checklist

- [x] I could not find a solution in the existing issues, docs, nor discussions
- [x] I have joined the [ZITADEL chat](https://zitadel.com/chat)

### Describe the docs your are missing or that are wrong

The documentation says [CockroachDB is preferred](https://zitadel.com/docs/self-hosting/manage/production#prefer-cockroachdb) for self-hosting but, given zitadel.com announced that it [switched to using PG earlier this year](https://zitadel.com/blog/move-to-postgresql), it would be good to know which of the following statements Zitadel still holds to and why:
* _We recommend using CockroachDB..._ (Is this still the case despite zitadel.com's internal switch?)
* _...if you are concerned about multi-regional data locality, [the way to go is with CockroachDB](https://www.cockroachlabs.com/docs/stable/multiregion-overview.html)._ (It could be I don't understand what is mean by "multi-regional data locality", but it sounds like this directly opposes the concern in the blog post about "difficulties in assuring our customers that their data would exclusively reside in their designated regions,"  Hmm... maybe the key is the word "exclusively"?)
* _The indexes for the database are optimized using load tests from [ZITADEL Cloud](https://zitadel.com/), which runs with CockroachDB._  (is this still the case despite switching to PG for zitadel.com?)

**Comment by fforootd** [comment, marked as answer]
Hey, happy to share some light here.

We **recommend using PostgreSQL as a default database**, and it looks to me that we need to clean up our docs a little to make that more clear. Since it is open source, can be self-hosted, and has a managed service offering on many cloud providers, it feels like the most solid choice in many cases.

The only reasons I can think of where CRDB might be a better fit are when you are concerned with high availability settings, where CRDB is easier to operate, and in setups where you want to spread your data across multiple regions. However, I think in most cases, HA can be achieved with better latency with a PG, and the multi-region topic is not something many people will be too concerned about (at least not too our learnings). Even with PG, you can do business continuity across regions, BTW.

One thing that makes us a little worried about CRDB is also their recent pricing changes, which, to my take, are cumbersome to many people 😁
